Florida_Analysis_Py_File.py

```diff
@@ -0,0 +1,534 @@
+]
+other_states_deaths_copy = other_states_deaths.copy()
+
+other_states_deaths_copy["Overdose_Per_100K"] = (
+    other_states_deaths_copy["Deaths"] / other_states_deaths_copy["Population"]
+) * 100_000
+# calculating and displaying results for drug related deaths - year, state and county wise
+
+other_states_result = (
+    other_states_deaths_copy.groupby(["Year", "State", "County"])["Overdose_Per_100K"]
+    .sum()
+    .reset_index()
+)
+other_states_result.head()
+# calculating yearwise drug related deaths results for Alabama
+alabama_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Alabama"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+alabama_results_yearwise
+# calculating yearwise drug related deaths results for Georgia
+georgia_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Georgia"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+georgia_results_yearwise
+# calculating yearwise drug related deaths results for Mississippi
+mississippi_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Mississippi"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+mississippi_results_yearwise
+# calculating yearwise drug related deaths results for South Carolina
+south_carolina_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "South Carolina"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+south_carolina_results_yearwise
+# calculating yearwise drug related deaths results for Tennessee
+tennessee_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Tennessee"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+tennessee_results_yearwise
+# calculating and displaying yearwise results for drug related deaths in neighboring states around Florida (combined)
+
+other_states_results_combined_yearwise = other_states_deaths_copy.groupby("Year")[
+    "Overdose_Per_100K"
+].mean()
+other_states_results_combined_yearwise
+
+# these yearwise results have been calculated while doing the pre-post analysis of drug overdose deaths in Florida
+florida_results_yearwise
+florida_summary = pd.DataFrame(florida_result.describe()["Overdose_Per_100K"]).rename(
+    columns={"Overdose_Per_100K": "Overdoses per 100k Residents - Florida"}
+)
+other_summary = pd.DataFrame(
+    other_states_result.describe()["Overdose_Per_100K"]
+).rename(columns={"Overdose_Per_100K": "Overdoses Per 100K Residents - Control States"})
+summary_statistics = pd.concat([florida_summary, other_summary], axis=1)
+summary_statistics
+other_states_result_copy = other_states_result.copy()
+
+other_states_result_copy["Year Relative To Policy"] = other_states_result_copy[
+    "Year"
+].apply(lambda x: scale_years(x))
+
+# double check no nulls in "year relative to policy"
+
+assert other_states_result_copy["Year Relative To Policy"].isnull().sum() == 0
```

Comment:
Hey Annapurna, I can see that you use assert to closely check if there are na values in your subset, good jog. But you probably want to remove this in order to make your code clean and good-formatted.

Reply:
Okay. Point noted! Thanks. I will make sure to remove any such extraneous code.